### PR TITLE
tests: collect central metrics in bigquery

### PR DESCRIPTION
**Pull request by janisz**
### Description

This PR adds a functionality to collect central metrics in bigquery when debug dump is created at the end of tests.

```sql
SELECT metric,
       labels,
       value
FROM   `acs-san-stackroxci.ci_metrics.stackrox_central_metrics`
WHERE  metric LIKE '%registry%'
       AND labels LIKE '%type=quay%' 
```

```json
[{
  "metric": "registry_client_request_duration_seconds",
  "labels": "type=quay",
  "value": "0.13406114"
}, {
  "metric": "registry_client_requests_total",
  "labels": "code=401 method=head type=quay",
  "value": "2"
}, {
  "metric": "registry_client_requests_total",
  "labels": "code=401 method=get type=quay",
  "value": "4"
}, {
  "metric": "registry_client_requests_total",
  "labels": "code=200 method=head type=quay",
  "value": "3"
}, {
  "metric": "registry_client_requests_total",
  "labels": "code=302 method=get type=quay",
  "value": "3"
}, {
  "metric": "registry_client_requests_total",
  "labels": "code=200 method=get type=quay",
  "value": "12"
}]
```

- [x] CHANGELOG update is not needed
- [x] Documentation is not needed
### Testing
- [x] inspected CI results
#### Automated testing
- [x] modified existing tests
- [x] contributed **no automated tests**
#### How I validated my change
CI

### CODE REVIEW

File: scripts/ci/lib.sh
Comment (porridge) [inline scripts/ci/lib.sh:162]: Please pin this and set up updates via dependabot. Moving targets usually end up being a real PITA.

File: scripts/ci/lib.sh
Comment (porridge) [inline scripts/ci/lib.sh:172]: What's `metrics.csv`? Should this have been `$csv_output`?

**Review by porridge — CHANGES_REQUESTED** [review]

File: scripts/ci/lib.sh
Comment (janisz) [reply scripts/ci/lib.sh:162]: I'll add a todo as it's also used in other place https://github.com/stackrox/stackrox/blob/2333f8046ebacb348fe5ca33ee327cb0f5c3f69b/tests/e2e/run-scale.sh#L77-L80

Comment (janisz) [reply scripts/ci/lib.sh:162]: Maybe we should move it into stackrox :thinking:

File: scripts/ci/lib.sh
Comment (janisz) [reply scripts/ci/lib.sh:172]: Good catch

**Review by gavin-stackrox — CHANGES_REQUESTED** [review]

File: scripts/ci/lib.sh
Comment (porridge) [inline scripts/ci/lib.sh:173]: I can see the test-metrics .csv files have a header line with comma-separated column names.
I might be missing something, but it seems `--skip_leading_rows=1` in  `bq` invocation is expecting such header...

As an aside, if the oder of columns here is supposed to match some order specified elsewhere, a link would be of great help to future maintainers.

**Comment by porridge** [comment]
@janisz I merged a slightly different but compatible implementation of your changes to `scripts/ci/metrics.sh` so you can just drop that part to resolve the conflict.

File: Makefile
Comment (janisz) [inline Makefile:144]: We need to wait for

- https://github.com/stackrox/prometheus-metric-parser/pull/15

**Comment by gavin-stackrox** [comment]
The changes LGTM. It just needs the `output_dir` resolved & CI to pass. I issued a `/test gke-scale-tests` to get that job exercised. Can you update "Testing Performed"? Thanks.

File: scripts/ci/lib.sh
Comment (porridge) [inline scripts/ci/lib.sh:187]: Thinking aloud:
if simple join with commas was insufficient [here](https://github.com/stackrox/prometheus-metric-parser/pull/15/files), then simple split and join on/with commas is likely to be fragile here too, no?

Comment (janisz) [reply scripts/ci/lib.sh]: Good point I can replace it with sed and it should be much simpler.
Unfortunately I've just discovered this is not working for multiline csv rows :(

File: scripts/ci/lib.sh
Comment (janisz) [inline scripts/ci/lib.sh:191]:
```suggestion
```

**Comment by porridge** [comment]
Not sure if this is ready for review...
```
INFO: Tue Jul 16 14:49:07 UTC 2024: Processing metrics from central debug dump
ls: cannot access 'debug-dump/*.zip': No such file or directory
14:49:07: Exception raised in ['scripts/ci/lib.sh', 'process_central_metrics', 'debug-dump'], Command '['scripts/ci/lib.sh', 'process_central_metrics', 'debug-dump']' returned non-zero exit status 2.
```
Please let me know if it indeed is, @janisz

**Comment by porridge** [comment]
This seems to break some jobs:
```
15:28:01: Running post command: ['scripts/ci/lib.sh', 'process_central_metrics', 'debug-dump']
INFO: Tue Jul 16 15:28:01 UTC 2024: Processing metrics from central debug dump
ls: cannot access 'debug-dump/*.zip': No such file or directory
15:28:01: Exception raised in ['scripts/ci/lib.sh', 'process_central_metrics', 'debug-dump'], Command '['scripts/ci/lib.sh', 'process_central_metrics', 'debug-dump']' returned non-zero exit status 2.
15:28:01: Post failure in: ['scripts/ci/lib.sh', 'process_central_metrics', 'debug-dump']
****
**** 15:28:01: ERROR: post test failed [Post failed: exit 1]
****
```

**Comment by janisz** [comment]
@porridge which jobs? I thought I've fixed it with last commit? The only failed job was ocp-4-12-scanner-v4-tests but for a different reason. I've retest it and also run scale test

File: scripts/ci/lib.sh
Comment (porridge) [inline scripts/ci/lib.sh:191]: This is for debugging, right? Can you remove the `cat` before merging?

**Review by porridge — APPROVED** [review]

**Comment by porridge** [comment]
> @porridge which jobs? I thought I've fixed it with last commit? The only failed job was ocp-4-12-scanner-v4-tests but for a different reason. I've retest it and also run scale test

Ah, maybe these were the stale 4.15 job results. I see they are gone now so 👍🏻

**Pull request merged**
